save-common/src/commonMain/kotlin/org/cqfn/save/core/config/TestConfig.kt

```diff
@@ -74,6 +75,38 @@ data class TestConfig(
         return listOf(this) + this.childConfigs.flatMap { it.getAllTestConfigs() }
     }
 
+    /**
+     * Walk all descendant configs and merge them with their parents
+     *
+     * @param createPluginConfigList a function which can create a list of [PluginConfig]s for this [TestConfig]
+     * @return an update this [TestConfig]
+     */
+    fun withEvaluatedDescendantConfigs(createPluginConfigList: (TestConfig) -> List<PluginConfig>): TestConfig {
+        getAllTestConfigs().forEach { testConfig ->
```

Comment:
Do we need `getAllTestConfigs` here for save-cloud? And can't move this function into `forEach` from Save.kt?
Now we goes through all configs twice

Reply:
Good point about going twice, I'll try to reconsider it. The reason for this was that in save-cloud we now have two steps: at first loop over all `GeneralConfig`s to get test suite names, and then loop over all active plugins to get a full list of tests.

Reply:
Removed the loop from this method and moved the call into the loop in `Save.kt`. I'll call `getAllTestConfigs().forEach` in save-cloud too